[CS2113-T13-1] MoneyMoover

woowenjun99 opened the pull request:
MoneyMoover is a CLI application for managing and transferring international currencies, optimized for use via a Command Line Interface (CLI) while still having the features of other money management applications.

ghzr0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
+is then extracted using saveMap, which filters out the rates for our supported currencies and performs type conversion.
+The savedMap attribute of ExchangeRates is set to this filtered map, which is then passed to Forex via getExchangeRates.
+
+If onFalire() is called, it means an unexpected error was encountered, such as losing Internet connection.
```
Is the spelling of the method wrong? (onFailure() instead of onFalire() )

ghzr0 replied:
Is the spelling of the method wrong? (onFailure() instead of onFalire() )

ghzr0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
 
-## Design & implementation
+The following sequence diagram shows how the Exchange command works
+![ExchangeSeqDiagram](images/ExchangeSeqDiagram.png)
```
Very detailed sequence diagram, but one point: is the spelling wrong? (underlined in red : BigDecimal instead of BigDecinmal)
![image](https://user-images.githubusercontent.com/88625158/228435043-3e2bb5b6-beed-42ab-8c32-f14dddb53142.png)

ghzr0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
+| v2.0    | user     | find a to-do item by name | locate a to-do without having to go through the entire list |
 
-## Non-Functional Requirements
+### Non-Functional Requirements
```
Please add more details in the next iteration of DG.

ghzr0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
 {Give non-functional requirements}
 
-## Glossary
+### Glossary
```
Please add more details in the next iteration of DG.

ghzr0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
+- _glossary item_ - Definition
 
-## Instructions for manual testing
+## Appendix: Instructions for manual testing
```
Please add more details in the next iteration of DG.

ghzr0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
+functionality is to view the balance of a specific currency if the currency is specified, else view all the currencies
+in the account.
+
+![BalanceSequenceDiagram](images/BalanceSeqDiagram.png)
```
Very detailed sequence diagram that showcases the balance feature

ghzr0 reviewed (commented):
Week 11 Tutorial DG review

nghochi123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
+
+The API of this component is specified in the `Ui.java`.
+
+![UI Class Diagram](images/UiClassDiagram.png)
```
Is the text on the enumeration part of the image too flushed to the left? Should there be some padding instead?

![image](https://user-images.githubusercontent.com/35862661/228457313-f82b30b2-c4ba-41d0-8e4d-449836205d28.png)

nghochi123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
 # Developer Guide
 
```
Very comprehensive sequence diagrams that show the flow of the application well.

nghochi123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
+Only currency account which 0 balance can be deleted.
+
+Given below is an example of the usage of this feature and the mechanism at each step
+
```
Should there be a space between the end of the text and the image? For Step 2 the image is in between the text which might not be ideal.

![image](https://user-images.githubusercontent.com/35862661/228458969-9459a57a-9e52-4cfe-875f-e8024d73d978.png)

nghochi123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
+rates from an online source. Future implementation will use an API to maintain up-to-date
+exchange rates.
+
+Exchange rate source: https://www.xe.com/currencyconverter/convert
```
Would it be better if this was in the form of a hyperlink instead? Such as:

Exchange Rate Link: `[source](https://www.xe.com/currencyconverter/convert)` which would give [source](https://www.xe.com/currencyconverter/convert).

Similar idea for the next subsection

nghochi123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,335 @@
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
Could add more details in the next iteration of DG (also for the appendix sections).

Gan868611 committed:
Update UG

Gan868611 committed:
Fix bug

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Update the help documentation
  Update help message and error message for transaction command errors
  Update help message to include transaction command
  Update the help command
  Fix bug to only accept amount greater or equal to 0.01
  Fix minor errors in sequence diagram
  Fix small error in transaction instantiation
  Remove unused import in Duke
  Add documentation in DG and UG for transactions

Gan868611 committed:
Merge branch 'master' into Update-UG

# Conflicts:
#	docs/UserGuide.md

jacob-stein1 committed:
Fix typo

jacob-stein1 committed:
Merge pull request #111 from jacob-stein1/default-rates

Add default rates in case of API failure

Gan868611 committed:
Remove duplicated code

Gan868611 committed:
Fix bugs

woowenjun99 committed:
Merge pull request #112 from Gan868611/Update-UG

Update ug

Gan868611 committed:
test

woowenjun99 committed:
Merge pull request #113 from Gan868611/Bug-fix-decimalRounding

Bug fix decimal rounding

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  test
  Fix bugs
  Remove duplicated code
  Fix typo
  Fix bug
  Update UG
  Use default exchange rates if API fails for any reason

jacob-stein1 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

Bawfen committed:
Add the ability to write to file.
An interface is used to define the functions for the Store
so that a dummy store can be used in unit tests.

woowenjun99 committed:
Update the error message

Bawfen committed:
Merge from master

woowenjun99 committed:
Complete the edit

Bawfen committed:
Return early from save if there is an exception

Bawfen committed:
Fix logging message in TransactionManager

Bawfen committed:
Update API Key

Bawfen committed:
Fix bug with LocalDateTimeAdapter

woowenjun99 committed:
Merge pull request #115 from Bawfen/Add-Persistence

Add the ability to write to file.

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Fix bug with LocalDateTimeAdapter
  Update API Key
  Fix logging message in TransactionManager
  Return early from save if there is an exception
  Add the ability to write to file. An interface is used to define the functions for the Store so that a dummy store can be used in unit tests.

Bawfen committed:
Fix issue with balances being saved and loaded wrongly

Bawfen committed:
Merge pull request #117 from Bawfen/Add-Persistence

Fix issue with balances being saved and loaded wrongly

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Fix issue with balances being saved and loaded wrongly

Bawfen committed:
Merge pull request #116 from woowenjun99/fix-up-the-dg

Delete messages

Bawfen committed:
Fix typo in DeveloperGuide

Bawfen committed:
Merge pull request #118 from Bawfen/Fix-DG

Fix typo in DeveloperGuide

kdh3799 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+
+![AccountListObjectDiagram4](images/AccountListObjectDiagram4.png)
+
+The following sequence diagram shows how the Create Account operation works
```
Maybe it is better to align the starting point of `execute` at the start of the program and the ending point of final return dashed line at the end of the program  to be on the same vertical line.
![image](https://user-images.githubusercontent.com/16986946/228762947-ad45a97c-21e9-4a3c-bd5d-bb556a5d2ceb.png)

kdh3799 replied:
Should the CreateAccountCommand be deactivated or destroyed once it returns?
![image](https://user-images.githubusercontent.com/16986946/228763304-1c7905e2-3f98-4573-bfe1-0c1c8ac76b79.png)

kdh3799 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+account balance.
+![AddWithdrawCommandObjectDiagram3](images/AddWithdrawCommandObjectDiagram3.png)
+
+The following sequence diagram shows how the add money operation works.
```
Should :AddCommand be activated at the same time when execute() function calls it?

![image](https://user-images.githubusercontent.com/16986946/228763600-9791b7a1-41fb-4612-8903-77493eabe7d0.png)

kdh3799 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+
+The following sequence diagram shows how the money withdrawal operation works.
+
+![](images/WithdrawCommandSeqDiagram.png)
```
Should :WithdrawCommand be activated at the same time when execute() function calls it?

![image](https://user-images.githubusercontent.com/16986946/228764064-0f1f10c9-f838-4f06-a65b-37e79c4d5b4e.png)

kdh3799 replied:
Should WIthdrawCommand be deactivated at the same time when it returns at the end?
![image](https://user-images.githubusercontent.com/16986946/228764627-3468f9b0-cfaa-4208-98e8-18161303debc.png)

kdh3799 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+- The relevant transactions are printed
 
-## Design & implementation
+The following sequence diagram shows how the Transaction command works
```
Should there be a return dashed line at the end of activation bar of TransactionCommand?
![image](https://user-images.githubusercontent.com/16986946/228765582-e8a6a4a7-53f9-43f5-8c91-70bef334bfc5.png)

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+        - [User Stories](#user-stories)
+        - [Non-Functional Requirements](#non-functional-requirements)
+        - [Glossary](#glossary)
+    - [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
```
Clear tables of content!

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+
+### Setting up the project in your computer
+
+First, **fork** this repo, and **clone** the fork into your computer.
+If you plan to use Intellij IDEA (highly recommended):
+
+1. **Configure the JDK**: Follow the guide [_[se-edu/guides] IDEA: Configuring the
+   JDK_](https://se-education.org/guides/tutorials/intellijJdk.html) to ensure Intellij is configured to use **JDK 11
+   **.
+2. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle
+   project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into
+   IDEA.<br>
+   :exclamation: Note: Importing a Gradle project is slightly different from importing a normal Java project.
+3. **Verify the setup**:
+    1. Run the `seedu.duke.Duke` and try a few commands.
+    2. Run the tests using `./gradlew check` to ensure they all pass.
```
Clear instructions on how to run start the application

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+
+The diagram above provides a high-level overview of how the project is structured. The main components are:
+
+1. The `Main` class which initialises all the other components at startup, and connects them with each other
+2. The `UI` component which is responsible for all user input and output
+3. The `Parser` component which parses user input and creates the relevant Command objects
+4. The `Command` component which executes the logic
+5. The `Accounts` component which manages the user's accounts
+6. The `Forex` (Foreign Exchange) component which handles exchange-rate related logic
+7. The `Transactions` component which manages the user's transactions
+8. The `Storage` component which handles the saving and loading of data to disk
```
Good overview for clarity on diagram

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+
+- Reads in the user input as `String`.
+
+### Parser component
+
+![Parser Class Diagram](images/ParserClassDiagram.png)
+
+The `Parser` Component
+
+- Parses the user input and creates the relevant `Command` object
+- Makes use of the `CommandType` enum to determine the type of command to create
+
```
I like how the guide was broken into components with each block contain greater details

ngyongjian commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
```
Could you consider adding more details to this section? (similarly for "Non-functional requirements", "Glossary" and "Appendix: Instructions for manual testing")

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+
+Step 3. The user passes command `withdraw` (eg. `withdraw SGD 25`), where `AMOUNT` must be smaller than the currency
+account balance.
+![AddWithdrawCommandObjectDiagram3](images/AddWithdrawCommandObjectDiagram3.png)
```
In general, the diagrams are great so far, but when viewed on the DG, it is very huge on screen. Maybe you can consider scaling down the image size for better readability.

ngyongjian commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+
+The API of this component is specified in the `Ui.java`.
+
+![UI Class Diagram](images/UiClassDiagram.png)
```
Could you consider editing this diagram? There should be some spacings at the left part.
<img width="394" alt="image" src="https://user-images.githubusercontent.com/91600220/229007698-ce37514f-98c3-4cc6-83c8-535b13d28829.png">

HiIAmTzeKean reviewed (commented):
Overall a comprehensive DG.
You can consider scaling images with this code below 
`<img src="drawing.jpg" alt="drawing" width="200"/>`

ngyongjian commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+| v2.0    | user     | find a to-do item by name | locate a to-do without having to go through the entire list |
 
-## Non-Functional Requirements
+### Non-Functional Requirements
```
Can consider adding in more details. Already mentioned above.

ngyongjian commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
 # Developer Guide
```
Overall very comprehensive class diagrams and sequence diagrams.

ngyongjian commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+rates from an online source. Future implementation will use an API to maintain up-to-date
+exchange rates.
+
+Exchange rate source: https://www.xe.com/currencyconverter/convert
```
Could you consider making this a link instead of pasting it as text?
<img width="719" alt="image" src="https://user-images.githubusercontent.com/91600220/229009881-607ef718-48ce-473b-b304-7213e0747012.png">

ngyongjian commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+transaction activity. A debit transaction will reflect on initial account and credit transaction will
+reflect on target account
+
+Exchange rate source: https://www.xe.com/currencyconverter/convert
```
Similarly as mentioned above, can consider making this a link instead of text.

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Fix typo in DeveloperGuide
  Complete the edit
  Update the error message

Gan868611 committed:
Update UG

Gan868611 committed:
Update UG

Gan868611 committed:
Edit format of indentation

Gan868611 committed:
Modify command summary table

Arif-Khalid committed:
Fix the exit command not immediately terminating the program

exetr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+The following is a high-level sequence of a single `create-account SGD` command, which demonstrates how the components
+interact with each other:
+
+![BasicSequence](images/BasicSequence.png)
```
Minor comment, but do consider whether you want to include the user entity within the diagram, as you have already stated that the command issued in question is `create-account SGD`, omitting the parts involving the user entity can help to simplify things

exetr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+
+Here is a class diagram of the `Accounts` component
+
+![AccountListClassDiagram](images/AccountListClassDiagram.png)
```
Consider whether you want to include all exceptions in this class diagram? Removing it can help to simplify the diagram, furthermore, inclusion of exceptions does not seem to add siginficant value to the discussion here. Perhaps it can be shifted to another separate diagram. Good use of composition to represent the relationship between AccountList and Account.

Bawfen committed:
Merge pull request #178 from Arif-Khalid/arifKhalid-FixExitCommand

Fix the exit command not immediately terminating the program

exetr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+- Exchange rates are manually pulled from https://www.xe.com/currencyconverter/convert
+- Each `Forex` instance must have two `Currency` associated with it
+
+### Open Exchange Rates API
```
Another minor comment, consider elaborating on how one can set up the program to interact with the API, such as specifying where and how to specify API tokens.

exetr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+functionality is to view the balance of a specific currency if the currency is specified, else view all the currencies
+in the account.
+
+![BalanceSequenceDiagram](images/BalanceSeqDiagram.png)
```
While I can understand the overall logic shown by the sequence diagram, you may want to consider omitting the interactions with the UI entity to help to simplify this diagram as it can appear to be very complex. Specifically, the presence of the UI entity does not indicate any extremely ikmportant functions (unless you would want to keep the `toString()` function call).

exetr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+- The Ui will print the unit rate both ways if no amount was provided
+
+The following sequence diagram shows how the Show Rate command works
+![ShowRateSeqDiagram](images/ShowRateSeqDiagram.png)
```
Double check this, but it seems like the conditions for the `alt` blocks are missing in this diagram.

exetr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
+- The new balances are printed
+
+The following sequence diagram shows how the Exchange command works
+![ExchangeSeqDiagram](images/ExchangeSeqDiagram.png)
```
If your inner `alt` block only consists of an if condition and not the else part, consider changing it to an `opt` block.

exetr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,361 @@
 
-## Design & implementation
+The following sequence diagram shows how the Transaction command works
+![TransactionSeqDiagram](images/ShowTransactionsSeqDiagram.png)
```
Consider whether it is strictly necessary to show the exception handling part, as to my knowledge, you are only indicating that a message is printed if an exception is met. Omitting it would help to simplify the diagram without losing its core purpose.

exetr reviewed (commented):
Great job on the DG so far, very good use of the different diagrams! I have left some comments for you to consider, mainly focused on simplifing certain sequence diagrams.

jacob-stein1 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

Gan868611 committed:
Reformat command summary

Gan868611 committed:
Modify exception message

Gan868611 committed:
Fix coding style

Gan868611 committed:
Add notes

Gan868611 committed:
Fix bugs

Gan868611 committed:
Fix bugs

Gan868611 committed:
Fix bugs

jacob-stein1 committed:
Add new limitations on show-rate to UG

jacob-stein1 committed:
Add min and max value to show-rate

jacob-stein1 committed:
Add confirmation message for API initialization

jacob-stein1 committed:
Change API key

jacob-stein1 committed:
Revert pause to 5 seconds for API call

jacob-stein1 committed:
Expand show-rate testing for new boundary checks

jacob-stein1 committed:
Remove float as intermediary in show-rate parsing

jacob-stein1 committed:
Specify missing accounts with exchange command

woowenjun99 committed:
Fix bug

jacob-stein1 committed:
Outline for jacob-stein1 PPP

jacob-stein1 committed:
Add API contribution

Arif-Khalid committed:
Merge remote-tracking branch 'AY2223S2-CS2113-T13-1/master'

Arif-Khalid committed:
Fixed too precise amount issues and error in detecting wrong dates

Arif-Khalid committed:
Fix checkstyle issues in commands

jacob-stein1 committed:
Finish JS PPP

jacob-stein1 committed:
Update JS PPP

jacob-stein1 committed:
Cut PPP text

Arif-Khalid committed:
Add DAYS_IN_MONTH constant hashmap for clarity

Arif-Khalid committed:
Fix checkstyle errors in DateConstants

Arif-Khalid committed:
WIP on PPP

Arif-Khalid committed:
Add Draft for PPP

Arif-Khalid committed:
Update AboutUs

Arif-Khalid committed:
Reduce length of PPP

Arif-Khalid committed:
Remove redundant log file

Bawfen committed:
Merge pull request #182 from Gan868611/Wrong-WithdrawComannd-Exception-desc

Wrong withdraw comannd exception desc

Bawfen committed:
Merge pull request #186 from woowenjun99/170-pe-dtester-a-cannot-tackle-extra-spaces

170-pe-dtester-a-cannot-tackle-extra-spaces

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Fix bug
  Fix coding style
  Modify exception message
  Fix the exit command not immediately terminating the program

Gan868611 committed:
Merge branch 'master' into Limit-description-100char

# Conflicts:
#	src/main/java/seedu/duke/commands/WithdrawCommand.java

jacob-stein1 committed:
Update JS PPP

Gan868611 committed:
Edit description message

Arif-Khalid committed:
Merge remote-tracking branch 'AY2223S2-CS2113-T13-1/master'

Gan868611 committed:
Merge branch 'master' into Update-UG-PED

jacob-stein1 committed:
Add comments clarifying exceptions

jacob-stein1 committed:
Merge branch 'master' into show-rate-ped

Gan868611 committed:
Limit desc to 100 char

Bawfen committed:
Merge pull request #184 from jacob-stein1/show-rate-ped

Show rate ped

Arif-Khalid committed:
Merge branch 'master' into arifKhalid-FixPEDryRunIssues

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Add comments clarifying exceptions
  Remove float as intermediary in show-rate parsing
  Expand show-rate testing for new boundary checks
  Revert pause to 5 seconds for API call
  Change API key
  Add confirmation message for API initialization
  Add min and max value to show-rate
  Add new limitations on show-rate to UG

Gan868611 committed:
Merge branch 'master' into Limit-description-100char

# Conflicts:
#	src/main/java/seedu/duke/constants/ErrorMessage.java

Bawfen committed:
Merge pull request #188 from Arif-Khalid/arifKhalid-FixPEDryRunIssues

Arif khalid fix pe dry run issues

jacob-stein1 committed:
Fix exception naming convention

jacob-stein1 committed:
Merge branch 'master' into exchange-cmd-ped

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Fix checkstyle errors in DateConstants
  Add DAYS_IN_MONTH constant hashmap for clarity
  Fix checkstyle issues in commands
  Fixed too precise amount issues and error in detecting wrong dates

Bawfen committed:
Merge pull request #185 from jacob-stein1/exchange-cmd-ped

Exchange cmd ped

Gan868611 committed:
Merge branch 'master' into Limit-description-100char

# Conflicts:
#	src/main/java/seedu/duke/commands/AddCommand.java
#	src/main/java/seedu/duke/commands/WithdrawCommand.java

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Fix exception naming convention
  Specify missing accounts with exchange command

Gan868611 committed:
Merge branch 'master' into Limit-description-100char

jacob-stein1 committed:
Merge pull request #190 from Arif-Khalid/arifKhalid-PPP

Arif Khalid PPP

jacob-stein1 committed:
Merge pull request #183 from Gan868611/Limit-description-100char

Limit description 100char

Gan868611 committed:
Merge branch 'master' into Update-UG-PED

# Conflicts:
#	docs/UserGuide.md

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Edit description message
  Remove redundant log file
  Reduce length of PPP
  Update AboutUs
  Add Draft for PPP
  WIP on PPP
  Fix bugs
  Fix bugs

Gan868611 committed:
Merge branch 'master' into Update-UG-PED

Gan868611 committed:
Fix bugs

jacob-stein1 committed:
Merge pull request #177 from Gan868611/Update-UG-PED

Update ug PED

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Fix bugs
  Limit desc to 100 char
  Fix bugs
  Add notes
  Reformat command summary
  Modify command summary table
  Edit format of indentation
  Update UG
  Update UG

jacob-stein1 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp into jacob-stein-ppp

jacob-stein1 committed:
Add photo to AboutUs

jacob-stein1 committed:
Correct path for portfolio link

woowenjun99 committed:
Add in a trim to ensure consistency in the input

jacob-stein1 committed:
Fix issue with rendering extracts

woowenjun99 committed:
Adjusted the error message

Arif-Khalid committed:
Merge remote-tracking branch 'AY2223S2-CS2113-T13-1/master'

Arif-Khalid committed:
Fix error in Transaction Command UG documentation

jacob-stein1 committed:
Merge pull request #194 from Arif-Khalid/arifKhalid-FixErrorInUG

Arif khalid fix error in ug

jacob-stein1 committed:
Merge pull request #192 from jacob-stein1/jacob-stein-ppp

Jacob stein ppp

jacob-stein1 committed:
Merge pull request #193 from woowenjun99/187-bug-withdraw-and-add-commands-follows-different-behaviour-when-no-amount-is-provided-1

187 bug withdraw and add commands follows different behaviour when no amount is provided 1

Bawfen committed:
Add check for gson parse errors and creates a new file if any are encountered.

Bawfen committed:
Move unknown command error message to ErrorMessage for consistency

Bawfen committed:
Update UserGuide Notes section with storage related information

Bawfen committed:
Adds default values if the balances in the locally saved files are invalid (above UPPER_BOUND or negative)

Bawfen committed:
Add PPP and image

Bawfen committed:
Merge pull request #197 from Bawfen/PPP-Ben

woowenjun99 committed:
Remove unused variables

woowenjun99 committed:
Create an InvalidBigDecimalException

woowenjun99 committed:
Create a validator class

woowenjun99 committed:
Integrate the validator class into the 2 commands

woowenjun99 committed:
Update the AddCommand and WithdrawCommand

woowenjun99 committed:
Update the value of the validator class

woowenjun99 committed:
Integrate in the error message into the validator class

woowenjun99 committed:
Make transactions final

woowenjun99 committed:
Drafting the PPP

woowenjun99 committed:
Update docs

woowenjun99 committed:
Update PPP with the sequence diagram

woowenjun99 committed:
Add final touches to the PPP

jacob-stein1 committed:
Merge pull request #198 from woowenjun99/153-ped

153 ped

jacob-stein1 committed:
Merge pull request #199 from woowenjun99/wen-jun-ppp

Wen jun ppp

jacob-stein1 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

jacob-stein1 committed:
Implement validator in exchange

jacob-stein1 committed:
Implement validator in show-rate

Bawfen committed:
Merge pull request #200 from jacob-stein1/implement-validator

Implement validator

Bawfen committed:
Merge from master

Bawfen committed:
Made UPPER_BOUND and LOWER_BOUND in Validator final to pass code style checks

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp: (27 commits)
  Implement validator in show-rate
  Implement validator in exchange
  Add final touches to the PPP
  Update PPP with the sequence diagram
  Update docs
  Drafting the PPP
  Make transactions final
  Integrate in the error message into the validator class
  Update the value of the validator class
  Update the AddCommand and WithdrawCommand
  Integrate the validator class into the 2 commands
  Create a validator class
  Create an InvalidBigDecimalException
  Remove unused variables
  Add PPP and image
  Fix error in Transaction Command UG documentation
  Adjusted the error message
  Fix issue with rendering extracts
  Add in a trim to ensure consistency in the input
  Correct path for portfolio link
  ...

Gan868611 committed:
Merge branch 'master' into Update-UG-PED

Bawfen committed:
Modify all Currency enum checks to uppercase by default, allowing users to enter in lowercase or uppercase

Gan868611 committed:
Update help message

Bawfen committed:
Rename all packages to com.moneymoover and replaces all mentions of Duke with MoneyMoover

Gan868611 committed:
create PPP

Gan868611 committed:
rebuild pages

Gan868611 committed:
fix bug

Gan868611 committed:
fix bugs

Bawfen committed:
Merge pull request #202 from Bawfen/147-currency-uppercase

147 currency uppercase

Gan868611 committed:
Merge branch 'master' into Update_developer_guide

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/images/AccountListObjectDiagram1.png
#	docs/images/AccountListObjectDiagram2.png
#	docs/images/AccountListObjectDiagram3.png
#	docs/images/CreateAccountSeqDiagram.png

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Rename all packages to com.moneymoover and replaces all mentions of Duke with MoneyMoover
  Modify all Currency enum checks to uppercase by default, allowing users to enter in lowercase or uppercase

Gan868611 committed:
Merge branch 'master' into Update-UG-PED

Gan868611 committed:
Merge branch 'master' into Update-DG

woowenjun99 committed:
Merge pull request #201 from Gan868611/Update-UG-PED

LGTM

jacob-stein1 committed:
Merge pull request #203 from Gan868611/Gan-Zhen-Yang-PPP

Gan zhen yang ppp

Bawfen committed:
Merge from master

Bawfen committed:
Modify UG help to be the same as the output of the help command

Bawfen committed:
Merge pull request #195 from Bawfen/119-program-crashes-if-the-json-files-are-corrupted

119 program crashes if the json files are corrupted

Bawfen committed:
Add section on currency and date related inputs to the UG

Gan868611 committed:
Fix sequence diagram

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Modify UG help to be the same as the output of the help command
  fix bugs
  fix bug
  rebuild pages
  create PPP
  Update help message
  Made UPPER_BOUND and LOWER_BOUND in Validator final to pass code style checks
  Adds default values if the balances in the locally saved files are invalid (above UPPER_BOUND or negative)
  Update UserGuide Notes section with storage related information
  Move unknown command error message to ErrorMessage for consistency
  Add check for gson parse errors and creates a new file if any are encountered.

woowenjun99 committed:
Merge pull request #205 from Bawfen/AddDateInputToUG

LGTM

Bawfen committed:
Merge pull request #204 from Gan868611/Update-DG

Update dg

Bawfen committed:
Remove files in wrong directory

woowenjun99 committed:
Merge pull request #210 from Bawfen/RemoveRedundantFiles

LGTM

woowenjun99 committed:
Add in the message for balances

woowenjun99 committed:
Refactor getAccounts

jacob-stein1 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

jacob-stein1 committed:
Remove ability to exchange between same currency

jacob-stein1 committed:
Add descriptions for exceptions that do not have one

jacob-stein1 committed:
Add JUnit test for same currency exception

jacob-stein1 committed:
Add assertion checks to API and commands using validator

woowenjun99 committed:
Merge pull request #214 from jacob-stein1/remove-exchange-between-same-currency

Remove exchange between same currency

Bawfen committed:
Merge pull request #213 from woowenjun99/issue-206-better-description-for-206

Add in the message for balances

Bawfen committed:
Add spacing to Message enum

Bawfen committed:
Add NFR, User Stories, Acknowledgements and Instructions for updating the API key

Bawfen committed:
Fix typo

jacob-stein1 committed:
Merge pull request #215 from Bawfen/FixDG

Fix dg

Bawfen committed:
Fix typo in Setting Up and hyperlink in Appendix

Bawfen committed:
Add the Instructions for Manual Testing section

jacob-stein1 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

jacob-stein1 committed:
Implement target exchange minimum

jacob-stein1 committed:
Add logging to API

jacob-stein1 committed:
Fix rounding error

Bawfen committed:
Merge pull request #217 from jacob-stein1/minimum-exchange-target

Minimum exchange target

Bawfen committed:
Merge from master

Bawfen committed:
Update API Key and PPP

Bawfen committed:
Add line about internet connection

Bawfen committed:
Fix bug where the amount loaded from file was wrong due to floating point calculation

Bawfen committed:
Uncommenting API key

Arif-Khalid committed:
Merge pull request #216 from Bawfen/ManualTestingInstructions

Manual testing instructions

Arif-Khalid committed:
Merge remote-tracking branch 'AY2223S2-CS2113-T13-1/master'

Arif-Khalid committed:
Fix standardisation of UG

Arif-Khalid committed:
Fix more standardisation errors in UG

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp: (22 commits)
  Uncommenting API key
  Fix bug where the amount loaded from file was wrong due to floating point calculation
  Add line about internet connection
  Update API Key and PPP
  Fix rounding error
  Add logging to API
  Implement target exchange minimum
  Add the Instructions for Manual Testing section
  Fix typo in Setting Up and hyperlink in Appendix
  Fix typo
  Add NFR, User Stories, Acknowledgements and Instructions for updating the API key
  Add spacing to Message enum
  Add assertion checks to API and commands using validator
  Add JUnit test for same currency exception
  Add descriptions for exceptions that do not have one
  Remove ability to exchange between same currency
  Refactor getAccounts
  Add in the message for balances
  Remove files in wrong directory
  Fix sequence diagram
  ...

Gan868611 committed:
Merge branch 'master' into Update-DG

Arif-Khalid committed:
Add further conditions for description in add and withdraw command

Gan868611 committed:
Modify seq diagram

Arif-Khalid committed:
Remove unnecessary spaces in UG

Arif-Khalid committed:
Fix formatting in example help command

Arif-Khalid committed:
Fix formatting in help command again

Arif-Khalid committed:
Fix to help command formatting missing spaces

Gan868611 committed:
Fix hyperlink

Arif-Khalid committed:
Fix spacing error in withdraw example command

Bawfen committed:
Merge pull request #226 from Arif-Khalid/arifKhalid-StandardiseDocumentation

Arif khalid standardise documentation in UG

Bawfen committed:
Change Appendix header to use hyphen instead of colon

Gan868611 committed:
Edit delete-account and create-account seq diagram

Arif-Khalid committed:
Fix Balance Sequence Diagram

Gan868611 committed:
Edit exchange seq diagram

woowenjun99 committed:
Update the Balance UG

woowenjun99 committed:
Update the PPP

Bawfen committed:
Add Storage Component to DG

woowenjun99 committed:
Merge branch 'master' into update-wen-ppp

Bawfen committed:
Fix to PPP

Bawfen committed:
Add Storage to TOC

Bawfen committed:
Fix links in DG

woowenjun99 committed:
Update the UI aspect of the class diagram

Bawfen committed:
Merge pull request #228 from woowenjun99/update-wen-ppp

Update wen ppp

Bawfen committed:
Merge pull request #229 from woowenjun99/update-ui-dg

Update the UI aspect of the class diagram

woowenjun99 committed:
Update the parser for the dg

Bawfen committed:
Merge pull request #230 from woowenjun99/update-parser-dg

Update the parser for the dg

Bawfen committed:
Fix BasicSequence

Bawfen committed:
Add return arrow to ShowTransactionsSeq

Bawfen committed:
Add colon for TransactionCommand instance

Arif-Khalid committed:
Fix errors in show rate sequence diagram

Gan868611 committed:
Modify add, withdraw seq diagram

woowenjun99 committed:
Update the account list dg

Bawfen committed:
Merge pull request #231 from woowenjun99/update-accounts-dg

Update the account list dg

Gan868611 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp:
  Update the account list dg
  Update the parser for the dg
  Update the UI aspect of the class diagram
  Update the PPP
  Update the Balance UG
  Fix spacing error in withdraw example command
  Fix to help command formatting missing spaces
  Fix formatting in help command again
  Fix formatting in example help command
  Remove unnecessary spaces in UG
  Add further conditions for description in add and withdraw command
  Fix more standardisation errors in UG
  Fix standardisation of UG

Arif-Khalid committed:
Add name to entity in show rate sequence diagram

Bawfen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T13-1/tp

Gan868611 committed:
Merge branch 'master' into Update-DG

# Conflicts:
#	docs/DeveloperGuide.md

Bawfen committed:
Merge from master

woowenjun99 committed:
Update the Forex component

Arif-Khalid committed:
Revert "Fix frames of balance diagram in front of activation boxes"

This reverts commit ac8e9cbc1367003d220aad2b358a25445e7829c6.

Arif-Khalid committed:
Fix frames of balance diagram in front of activation boxes

Bawfen committed:
Update ShowTransactionsSeqDiagram

Gan868611 committed:
Move alt frame to the back (all seq diagram)

woowenjun99 committed:
Update the ForexComponent

Arif-Khalid committed:
Fix frames of balance diagram in front of activation box

Bawfen committed:
Merge pull request #233 from woowenjun99/update-forex-dg

Update the Forex component

Arif-Khalid committed:
Fix frames of show rate sequence diagrams in front of activation box

Gan868611 committed:
Fix alt frame

Gan868611 committed:
delete extra files

Bawfen committed:
Merge pull request #227 from Gan868611/Update-DG

Update dg

Bawfen committed:
Update BasicSequence diagram

Bawfen committed:
Merge from master

Arif-Khalid committed:
Merge remote-tracking branch 'AY2223S2-CS2113-T13-1/master'

Arif-Khalid committed:
Merge branch 'master' into arifKhalid-DGShowRateAndBalanceCommand

jacob-stein1 committed:
Merge pull request #236 from Arif-Khalid/arifKhalid-DGShowRateAndBalanceCommand

Arif khalid dg show rate and balance command

Arif-Khalid committed:
Merge pull request #234 from Bawfen/AddStorageToDG

Add storage to dg

woowenjun99 committed:
Update the ppp

Bawfen committed:
Merge pull request #237 from woowenjun99/update-ppp

Update the ppp
